ceph/pkg/adapter/adapter.go

```diff
@@ -19,53 +19,102 @@ package adapter
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
-	"log"
 	"net/http"
 	"time"
 
-	cloudevents "github.com/cloudevents/sdk-go"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"go.uber.org/zap"
 	ceph "knative.dev/eventing-contrib/ceph/pkg/apis/v1alpha1"
-	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
 )
 
-var theClient cloudevents.Client
-var stopCh chan struct{}
+type envConfig struct {
+	adapter.EnvConfig
+
+	// Port to listen incoming connections
+	Port string `envconfig:"PORT" default:"8080"`
+}
+
+// cephReceiveAdapter converts incoming Ceph notifications to
+// CloudEvents and then sends them to the specified Sink
+type cephReceiveAdapter struct {
+	logger *zap.SugaredLogger
+	client cloudevents.Client
+	port   string
+}
+
+// NewEnvConfig function reads env variables defined in envConfig structure and
+// returns accessor interface
+func NewEnvConfig() adapter.EnvConfigAccessor {
+	return &envConfig{}
+}
+
+// NewAdapter returns the instance of cephReceiveAdapter that implements adapter.Adapter interface
+func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClient cloudevents.Client) adapter.Adapter {
+	logger := logging.FromContext(ctx)
+	env := processed.(*envConfig)
+
+	return &cephReceiveAdapter{
+		logger: logger,
+		client: ceClient,
+		port:   env.Port,
+	}
+}
+
+// Start the ceph bucket notifications to knative adapter
+func (ca *cephReceiveAdapter) Start(stopCh <-chan struct{}) error {
+	http.HandleFunc("/", ca.postHandler)
+	go http.ListenAndServe(":"+ca.port, nil)
+	ca.logger.Info("Ceph to Knative adapter spawned HTTP server")
+	<-stopCh
+
+	ca.logger.Info("Ceph to Knative adapter terminated")
+	return nil
+}
 
 // postMessage convert bucket notifications to knative events and sent them to knative
-func postMessage(notification ceph.BucketNotification) error {
+func (ca *cephReceiveAdapter) postMessage(notification ceph.BucketNotification) error {
 	eventTime, err := time.Parse(time.RFC3339, notification.EventTime)
 	if err != nil {
-		log.Printf("Failed to parse event timestamp, using local time. Error: %s", err.Error())
+		ca.logger.Infof("Failed to parse event timestamp, using local time. Error: %s", err.Error())
 		eventTime = time.Now()
 	}
 
-	event := cloudevents.NewEvent(cloudevents.VersionV1)
+	event := cloudevents.NewEvent(cloudevents.VersionV03)
 	event.SetID(notification.ResponseElements.XAmzRequestID + notification.ResponseElements.XAmzID2)
 	event.SetSource(notification.EventSource + "." + notification.AwsRegion + "." + notification.S3.Bucket.Name)
 	event.SetType("com.amazonaws." + notification.EventName)
 	event.SetDataContentType(cloudevents.ApplicationJSON)
```

Comment:
you don't need this because set data will already set the datacontenttype to `application/json`